lldb/test/API/functionalities/data-formatter/type-recognizer/value-object-cast/TestRecognizedValue.py

```diff
@@ -0,0 +1,514 @@
+"""
+Use lldb Python API to test ValueObject type recognizer
+"""
+
+
+import lldb
+from lldbsuite.test.decorators import *
+from lldbsuite.test.lldbtest import *
+from lldbsuite.test import lldbutil
+
+
+class RecognizedValueTestCase(TestBase):
+    def setUp(self):
+        # Call super's setUp().
+        TestBase.setUp(self)
+
+        # Execute the cleanup function during test case tear down.
+        self.addTearDownHook(self.cleanup)
+
+        # Find the line number to break for main.cpp.
+        self.br_line_cast_func = line_number(
+            "main.cpp", "// Break here in cast_func()."
+        )
+
+        # Prepare executable.
+        self.build()
+        self.exe = self.getBuildArtifact("a.out")
+
+    def test_find_variable(self):
+        """Test recognizing variables using the FindVariable."""
+        # Create a target from the debugger.
+        target = self.dbg.CreateTarget(self.exe)
+        self.assertTrue(target, VALID_TARGET)
+
+        # Set up our breakpoints:
+        br_cast_func = target.BreakpointCreateByLocation(
+            "main.cpp", self.br_line_cast_func
+        )
+        self.assertTrue(br_cast_func, VALID_BREAKPOINT)
+
+        # Now launch the process, and do not stop at the entry point.
+        process = target.LaunchSimple(None, None, self.get_process_working_directory())
+        self.assertState(process.GetState(), lldb.eStateStopped, PROCESS_STOPPED)
+
+        use_dynamic = lldb.eDynamicCanRunTarget
+        no_dynamic = lldb.eNoDynamicValues
+
+        # Okay now run to cast breakpoint:
+        threads = lldbutil.get_threads_stopped_at_breakpoint(process, br_cast_func)
+        self.assertEqual(len(threads), 1)
+        thread = threads[0]
+
+        frame = thread.GetFrameAtIndex(0)
+
+        # Get static "input_var".
+        var_static = frame.FindVariable("input_var", no_dynamic)
+        self.assertTrue(var_static)
+        self.assertTrue(var_static.type.name == "A_2 *")
+        self.assertTrue(var_static.IsValid())
+        self.assertFalse(var_static.IsDynamic())
+
+        # We need to make sure that lldb_private::ValueObjectRecognizedValue
```

Comment:
This first sentence misleads readers into thinking that in the asserts that follow we expect to see `B_2 *`, because that's what we want to make sure of. Instead, you're testing for `C_1 *`